pepper-apis/src/main/java/org/broadinstitute/ddp/service/actvityinstanceassembler/ActivityInstanceAssembleService.java

```diff
@@ -0,0 +1,188 @@
+import org.broadinstitute.ddp.model.activity.instance.answer.Answer;
+import org.broadinstitute.ddp.pex.PexInterpreter;
+import org.broadinstitute.ddp.pex.TreeWalkInterpreter;
+import org.broadinstitute.ddp.route.GetActivityInstanceRoute;
+import org.broadinstitute.ddp.service.actvityinstanceassembler.block.FormBlockCreator;
+import org.broadinstitute.ddp.service.actvityinstanceassembler.block.question.QuestionCreator;
+import org.broadinstitute.ddp.service.actvityinstanceassembler.block.question.ValidationRuleCreator;
+import org.jdbi.v3.core.Handle;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Provides creation of {@link ActivityInstance} an alternative way:
+ * instead of building it by fetching all data from DB
+ * it gets most of study data from {@link ActivityDefStore} and then fetch rest
+ * of data from DB (answers, validation messages).
+ *
+ * <p>Main method of this service is
+ * {@link #assembleActivityInstance(Handle, String, ContentStyle, String, String, String, ActivityInstanceDto)}
+ * which needs to be called for example from {@link GetActivityInstanceRoute}.
+ * In this method executed the following steps:
+ * <ul>
+ *     <li>find activity data in {@link ActivityDefStore} by study guid and activity instance dto;</li>
+ *     <li>if data is found then:<br>
+ *        - get answers from DB;<br>
+ *        - run the process of {@link ActivityInstance} building.
+ *     </li>
+ * </ul>
+ *
+ * <p><b>Activity instance building steps:</b>
+ * <ul>
+ *     <li>create {@link FormInstance} (via constructor, setting data from {@link ActivityInstanceDto});</li>
+ *     <li>iterate through {@link FormActivityDef#getAllSections()} and add sections to the {@link FormInstance};</li>
+ *     <li>in each {@link FormSectionDef} iterate through {@link FormSectionDef#getBlocks()} and add
+ *       blocks to an added {@link FormInstance} section;</li>
+ *     <li>in each {@link FormBlockDef} iterate through {@link FormBlockDef#getQuestions()} and add
+ *       questions to an added {@link FormInstance} block;</li>
+ *     <li>for each of added question find validations and add to the question;</li>
+ *     <li>for each of added question find answers and add to the question.</li>
+ * </ul>
+ *
+ * <p>For each element creation implemented creators (instances of interface {@link ElementCreator}.
+ * <br>Element creators hierarchy:
+ * <pre>
+ *   {@link FormInstanceCreator}
+ *     {@link FormSectionCreator}
+ *       {@link FormBlockCreator}
+ *         {@link SectionIconCreator}
+ *         {@link QuestionCreator}
+ *           {@link ValidationRuleCreator}
+ * </pre>
+ *
+ * <p>NOTE: it is defined a class {@link Context} used to pass the basic parameters to each element creator (so it no
+ * need to pass multiple parameter in each creator - only one parameter.
+ */
+public class ActivityInstanceAssembleService {
```

Comment:
Can we just put this in `ActivityInstanceService` instead of creating a new class?

Reply:
You mean to add this method to this class?
But I'd better to call from this method a method of a delegate class doing this logic, because `ActivityInstanceService` already too huge class and may be good to move out some code to delegate class.
I can rename `ActivityInstanceAssembleService` to `ActivityInstanceBuilder` and add to `ActivityInstanceService` a method:
```
ActivityInstance buildActivityInstance(...) {
   return new ActivityInstanceBuilder().build(...);
}
```
Would it be ok?